federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -963,4 +978,102 @@ private Stream<LDAPObject> paginatedSearchLDAP(LDAPQuery ldapQuery, Integer firs
     public String toString() {
         return "LDAPStorageProvider - " + getModel().getName();
     }
+
+    @Override
+    public void decorateUserProfile(UserProfileMetadata metadata) {
+        Predicate<AttributeContext> ldapUsersSelector = (attributeContext -> {
+            UserModel user = attributeContext.getUser();
+            if (user == null) {
+                return false;
+            }
+            return getModel().getId().equals(user.getFederationLink());
+        });
+
+        Predicate<AttributeContext> onlyAdminCondition = context -> metadata.getContext() == UserProfileContext.USER_API;
+
+        Map<String, Object> inputTextAnnotations = new HashMap<>();
+        inputTextAnnotations.put("inputType", "text");
```

Comment:
Ok, I will remove annotation from this. I used the annotation because of https://github.com/keycloak/keycloak/issues/24319, which render the fields as "Combobox" . But it is already fixed according to UI team, so I might just need to completely rebuild UI...

I think that if administrators want to change annotations for attributes, they may need to define the particular attribute in user-profile? With my current changes, in case that field is already defined in user-profile, it is not added by user-storage provider due to https://github.com/mposolda/keycloak/blob/23878-user-profile-federation-rebase/federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java#L1013-L1015

Reply:
Agreed.